src/app/configuration/environments/edit-environment/edit-environment.component.html

```diff
@@ -371,9 +387,23 @@ <h4>Environment Specifics</h4>
         </tr>
       </thead>
       <tbody>
-        <tr *ngFor="let item of env.environment_specifics | keyvalue">
-          <td>{{ item.key }}</td>
-          <td>{{ item.value }}</td>
+        <tr *ngFor="let item of env.environment_specifics | keyvalue">         
+          <!-- view of unedit value -->
+          <td *ngIf="uneditedEnv.environment_specifics.hasOwnProperty(item.key)">{{ item.key }}</td>
+          <td *ngIf="item.value == uneditedEnv.environment_specifics[item.key]">{{ item.value }}</td>  
+          <!-- view of new key -->
+          <td *ngIf="!uneditedEnv.environment_specifics.hasOwnProperty(item.key)" class="add-elem">{{ item.key }}</td>
+          <td *ngIf="!uneditedEnv.environment_specifics.hasOwnProperty(item.key)" class="add-elem">{{ item.value }}</td> 
+          <!-- view of edit value -->
+          <td *ngIf="uneditedEnv.environment_specifics.hasOwnProperty(item.key) && item.value != uneditedEnv.environment_specifics[item.key]">
+            <span class="del-elem arrow-after">{{ uneditedEnv.environment_specifics[item.key] }}</span > 
+            <span class="add-elem">{{ item.value }}</span > 
+          </td>
+        </tr>
+        <!-- view deleted key -->
+        <tr *ngFor="let item of uneditedEnv.environment_specifics | keyvalue">
+          <td *ngIf="!env.environment_specifics.hasOwnProperty(item.key)" class="del-elem">{{ item.key }}</td>       
```

Comment:
you are doing the same *ngIf twice.
maybe use 

`<ng-container` with the ngIf, and put the both  td elements inside it.